### DISCUSSION TITLE
can i replace h3 with express js in nuxt 3 ?

### DISCUSSION THREAD

**Discussion by rajeshtva** [nuxt/nuxt#16764] [Questions]
**I am learning express.js**. I am new to it. but i have worked with nuxt 2 a lot. there we had express js integration guide in official docs. but we don't have such thing in nuxt 3. I know nuxt 3 uses h3 instead of express. **So out of curiosity, is it possible to replace h3 with express js for apis only without starting another node.js process.**.. that will make learning express js easier and i will able to use nuxt 3.

**Comment by sevillaarvin** [comment]
Related https://github.com/nuxt/framework/discussions/1704

**Comment by TheAlexLichter** [comment]
My take on that: Without forking nitro and changing all the references (plus building some features h3 has but express hasn't) - no.

But luckily, h3 supports connect/express middleware for compatibility and you could also use it standalone 😋
So maybe try it the other way around and "learn" h3 (while keep using all the existing middleware of connect and epxress that you might want to use)?
